dio/lib/src/dio_mixin.dart

```diff
@@ -517,7 +517,10 @@ abstract class DioMixin implements Dio {
         stream,
         cancelToken?.whenCancel,
       );
-      final headers = Headers.fromMap(responseBody.headers);
+      final headers = Headers.fromMap(
+        responseBody.headers,
+        caseSensitive: reqOpt.caseSensitiveHeaders,
```

Comment:
I guess it makes sense to stick to one naming-pattern, so it should be always `preserveHeaderCase`.